src/libcore/alloc.rs

```diff
@@ -213,18 +213,21 @@ impl Layout {
     /// Creates a layout by rounding the size of this layout up to a multiple
     /// of the layout's alignment.
     ///
-    /// Returns `Err` if the padded size would overflow.
-    ///
     /// This is equivalent to adding the result of `padding_needed_for`
     /// to the layout's current size.
     #[unstable(feature = "alloc_layout_extra", issue = "55724")]
     #[inline]
-    pub fn pad_to_align(&self) -> Result<Layout, LayoutErr> {
+    pub fn pad_to_align(&self) -> Layout {
         let pad = self.padding_needed_for(self.align());
-        let new_size = self.size().checked_add(pad)
-            .ok_or(LayoutErr { private: () })?;
+        // This cannot overflow: it is an invariant of Layout that
+        // > `size`, when rounded up to the nearest multiple of `align`,
+        // > must not overflow (i.e., the rounded value must be less than
+        // > `usize::MAX`)
+        let new_size = self.size() + pad;
 
-        Layout::from_size_align(new_size, self.align())
+        // SAFETY: This necessarily respects the from_size_align
+        // prerequisites per the above.
+        unsafe { Layout::from_size_align_unchecked(new_size, self.align()) }
```

Comment:
I have not seen this code ever before... so if I am to review this, please use `from_size_align().unwrap()` and no unsafe code. I doubt this is performance-critical.

If you really want to avoid this run-time check, maybe @pnkfelix or @SimonSapin could review.